src/test/regress/expected/multi_transaction_recovery_multiple_databases.out

```diff
@@ -0,0 +1,364 @@
+       (:worker_2_group_id, :'transaction_2_worker_2_db_1_name');
+INSERT INTO pg_dist_transaction
+VALUES (:worker_1_group_id, 'citus_0_should_be_forgotten_' || :'db1_oid'),
+       (:worker_2_group_id, 'citus_0_should_be_forgotten_' || :'db1_oid');
+\c db2 - - :master_port
+INSERT INTO pg_dist_transaction
+VALUES (:worker_1_group_id, :'transaction_2_worker_1_db_2_name'),
+       (:worker_2_group_id, :'transaction_2_worker_2_db_2_name');
+INSERT INTO pg_dist_transaction
+VALUES (:worker_1_group_id, 'citus_0_should_be_forgotten_' || :'db2_oid'),
+       (:worker_2_group_id, 'citus_0_should_be_forgotten_' || :'db2_oid');
+\c db1 - - :master_port
+SELECT count(*) != 0
+FROM pg_dist_transaction;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT recover_prepared_transactions() > 0;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT count(*) = 0
+FROM pg_dist_transaction;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+\c db2 - - :master_port
+SELECT count(*) != 0
+FROM pg_dist_transaction;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT recover_prepared_transactions() > 0;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT count(*) = 0
+FROM pg_dist_transaction;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+\c regression - - :master_port
+SELECT count(pg_terminate_backend(pid)) > 0
+FROM pg_stat_activity
+WHERE pid <> pg_backend_pid()
+  AND datname = 'db1' ;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+DROP DATABASE db1;
+SELECT count(pg_terminate_backend(pid)) > 0
+FROM pg_stat_activity
+WHERE pid <> pg_backend_pid()
+  AND datname = 'db2' ;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+DROP DATABASE db2;
+\c - - - :worker_1_port
+SELECT count(pg_terminate_backend(pid)) > 0
+FROM pg_stat_activity
+WHERE pid <> pg_backend_pid()
+  AND datname = 'db1' ;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+DROP DATABASE db1;
+SELECT count(pg_terminate_backend(pid)) > 0
+FROM pg_stat_activity
+WHERE pid <> pg_backend_pid()
+  AND datname = 'db2' ;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+DROP DATABASE db2;
+\c - - - :worker_2_port
+-- Count of terminated sessions is not important for the test,
+-- it is just to make output predictable
+SELECT count(pg_terminate_backend(pid)) > 0
+FROM pg_stat_activity
+WHERE pid <> pg_backend_pid()
+  AND datname = 'db1' ;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+DROP DATABASE db1;
+SELECT count(pg_terminate_backend(pid)) > 0
```

Comment:
is there any risk of this being 0? maybe >= 0 if it's not important

(we sometimes run our tests with postgres running under valgrind, which creates all sorts of weird timings)

Reply:
Good point: since it is a clean up, essentially, it shouldn't matter. Let me do a quick fix...